helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -284,6 +285,21 @@ public void purgeOfflineInstances(String clusterName, long offlineDuration) {
     }
   }
 
+  private void purgeInstance(String clusterName, String instanceName) {
+    logger.info("Purge instance {} from cluster {}.", instanceName, clusterName);
+
+    String liveInstancePath = PropertyPathBuilder.liveInstance(clusterName, instanceName);
```

Comment:
Why do we need to check liveInstancePath here? For this scenario, you already checked live instance in findTimeoutOfflineInstances, right? So this function should do pure removing.
Moreover, with this assumption, you can just refactor the code to:
dropInstance() {
//check;
purgeInstance();
}
purgeInstance() {
// remove all ndoes.
}

Reply:
I adopted a similar style with the previous code, but I agree with the refactoring suggestion.